### DISCUSSION TITLE
Prefect ModuleNotFoundError when running from UI

### DISCUSSION THREAD

**Discussion by carlryn** [PrefectHQ/prefect#3311] [Q&A] [answered]
I'm following the Prefect tutorial available at: https://docs.prefect.io/core/tutorial/01-etl-before-prefect.html.
The code can be downloaded from the git: https://github.com/PrefectHQ/prefect/tree/master/examples/tutorial

The tutorials have a dependency to aircraftlib which is a directory under tutorials. I can execute the Flows through the terminal with:

    python 02_etl_...
and it executes perfectly!

I've created a project, and added the Flow to that project. Through the Prefect Server UI I can run the Flow, but it fails with the error message:

> State Message:
Failed to load and execute Flow's environment: ModuleNotFoundError("No module named 'aircraftlib'")

How should I handle the dependency when executing the Flows through the Prefect Server UI?

**Comment by joshmeek** [comment, marked as answer]
Hi @carlryn this tutorial currently is only set up for local execution without a backend (something we're looking to improve in the near future) and I can see where it would have issues in its current state. In order for local modules to be importable when a flow run is executed it either needs to be importable from the `PYTHONPATH` or packaged up with something like [Docker storage](https://docs.prefect.io/api/latest/environments/storage.html#docker)

**Reply by carlryn** [reply]
Thanks Josh!

Adding the following lines made the trick!

import sys
sys.path.insert(0, './aircraftlib')

Also, is this how you are expected to work with prefect?
Or do you have any good(perhaps advanced) examples of best practice?
